integration-v2.groovy

```diff
@@ -0,0 +1,825 @@
+#!groovy
+/*
+* Copyright (c) 2025 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* WSO2 Inc. licenses this file to you under the Apache License,
+* Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*
+*/
+
+import groovy.json.JsonSlurperClassic 
+
+// Input parameters
+String product = params.product
+String productVersion = params.productVersion
+String productDeploymentRegion = params.productDeploymentRegion
+String[] osList = params.osList?.split(',')?.collect { it.trim() } ?: []
+String[] jdkList = params.jdkList?.split(',')?.collect { it.trim() } ?: []
+String[] databaseList = params.databaseList?.split(',')?.collect { it.trim() } ?: []
+String albCertArn = params.albCertArn
+String acpUpdateLevel = params.acpUpdateLevel?: "-1"
+String tmUpdateLevel = params.tmUpdateLevel?: "-1"
+String gwUpdateLevel = params.gwUpdateLevel?: "-1"
+Boolean useStaging = params.useStaging
+String tfS3Bucket = params.tfS3Bucket
+String tfS3region = params.tfS3region
+String awsCred = params.awsCred
+String dbPassword = params.dbPassword
+String project = params.project?: "wso2"
+Boolean onlyDestroyResources = params.onlyDestroyResources
+Boolean destroyResources = params.destroyResources
+Boolean skipTfApply = params.skipTfApply
+Boolean skipDockerBuild = params.skipDockerBuild
+
+// Default values
+def deploymentPatterns = []
+String updateType = "u2"
+String hostName = ""
+String dbUser = "wso2carbon"
+// Helm repository details
+String helmRepoUrl = "https://github.com/wso2/helm-apim.git"
+String helmRepoBranch = "apim-intg"
+String helmDirectory = "helm-apim"
+// APIM Test Integration repository details
+String apimIntgRepoUrl = "https://github.com/wso2/apim-test-integration.git"
+String apimIntgRepoBranch = "4.5.0-profile-automation"
+String apimIntgDirectory = "apim-test-integration"
+String tfDirectory = "terraform"
+String tfEnvironment = "dev"
+
+String githubCredentialId = "WSO2_GITHUB_TOKEN"
+def dbEngineList = [
+    "mysql": [
+        version: "5.7",
```

Comment:
We need to move to MySQL 8.x with this. Shall we test with 8.x versions?

We might have to create the databases with latin1 collation in order to support this.